packages/local_auth/test/local_auth_test.dart

```diff
@@ -73,6 +73,30 @@ void main() {
         );
       });
 
+      test('authenticate with `localizedFallbackTitle` on iOS.', () async {
+        final iosAuthMessages =
+            IOSAuthMessages(localizedFallbackTitle: 'Enter PIN');
+        setMockPathProviderPlatform(FakePlatform(operatingSystem: 'ios'));
+        await localAuthentication.authenticate(
+          localizedReason: 'Needs secure',
+          biometricOnly: true,
+          iOSAuthStrings: iosAuthMessages,
+        );
+        expect(
+          log,
+          <Matcher>[
+            isMethodCall('authenticate',
+                arguments: <String, dynamic>{
+                  'localizedReason': 'Needs secure',
+                  'useErrorDialogs': true,
+                  'stickyAuth': false,
+                  'sensitiveTransaction': true,
+                  'biometricOnly': true,
+                }..addAll(iosAuthMessages.args)),
```

Comment:
I see that this is standard practice in the existing tests in this file, but it's actually a very bad idea; this test wouldn't have failed with your incorrect key bug, because it's using part of the object under test to generate the test expectations, which undermines the test. Please replace this with an explicit list of expected arguments. (You don't have to fix the existing instances since it's not related to your PR, but if you wanted to that would be great.)